libs/blocks/preflight/panels/seo.js

```diff
@@ -0,0 +1,223 @@
+import { html, signal, useEffect } from '../../../deps/htm-preact.js';
+
+const DEF_ICON = 'purple';
+const DEF_DESC = 'Checking...';
+
+const h1Result = signal({ icon: DEF_ICON, title: 'H1 count', description: DEF_DESC });
+const titleResult = signal({ icon: DEF_ICON, title: 'Title size', description: DEF_DESC });
+const canonResult = signal({ icon: DEF_ICON, title: 'Canonical', description: DEF_DESC });
+const descResult = signal({ icon: DEF_ICON, title: 'Meta description', description: DEF_DESC });
+const bodyResult = signal({ icon: DEF_ICON, title: 'Body size', description: DEF_DESC });
+const loremResult = signal({ icon: DEF_ICON, title: 'Lorem Ipsum', description: DEF_DESC });
+const linksResult = signal({ icon: DEF_ICON, title: 'Links', description: DEF_DESC });
+let Passed = true;
+
+function checkH1s() {
+  const h1s = document.querySelectorAll('h1');
+  const result = { ...h1Result.value };
+  if (h1s.length === 1) {
+    result.icon = 'green';
+    result.description = 'Only one H1 on the page.';
+  } else {
+    Passed = false;
+    result.icon = 'red';
+    if (h1s.length > 1) {
+      result.description = 'Reason: More than one H1 on the page.';
+    } else {
+      result.description = 'Reason: No H1 on the page.';
+    }
+  }
+  h1Result.value = result;
+}
+
+async function checkTitle() {
+  const titleSize = document.title.replace(/\s/g, '').length;
+  const result = { ...titleResult.value };
+  if (titleSize < 15) {
+    Passed = false;
+    result.icon = 'red';
+    result.description = 'Reason: Title size is too short.';
+  } else if (titleSize > 70) {
+    Passed = false;
+    result.icon = 'red';
+    result.description = 'Reason: Title size is too long.';
+  } else {
+    result.icon = 'green';
+    result.description = 'Title size is good.';
+  }
+  titleResult.value = result;
+}
+
+async function checkCanon() {
+  const canon = document.querySelector("link[rel='canonical']");
+  const result = { ...canonResult.value };
+  const { href } = canon;
+  const resp = await fetch(href, { method: 'HEAD' });
+  if (!resp.ok) {
+    Passed = false;
+    result.icon = 'red';
+    result.description = 'Reason: Error with canonical reference.';
```

Comment:
This set of three procedures happens a few times, could break out into a function:
```
failedPreflightCheck(passed, desc) {
    passed = false;
    result.icon = 'red';
    result.description = desc;
}
```